example/src/main/scala/xs4s/example/FindMostPopularWikipediaKeywordsZIOApp.scala

```diff
@@ -0,0 +1,56 @@
+package xs4s.example
+
+import zio._
+import zio.console._
+import zio.stream._
+import zio.blocking.Blocking
+import java.io.IOException
+import java.io.InputStream
+import java.net.URL
+import java.net.URLConnection
+import java.util.zip.GZIPInputStream
+import java.util.zip.InflaterInputStream
+import java.util.zip.Inflater
+import scala.collection.JavaConverters._
+import xs4s.ziocompat._
+import xs4s.syntax.zio._
+
+object FindMostPopularWikipediaKeywordsZIOApp extends zio.App {
+
+  def run(args: List[String]): URIO[ZEnv, ExitCode] = {
+
+    wikipediaXmlBytes
+      .via(byteStreamToXmlEventStream()(_))
+      .via(s => anchorExtractor.toZIOPipeThrowError(s) )
+      .via(stream => // by default, get a sub-set of the stream
+        if (args.contains("full")) stream else stream.take(500)
+      )
+      .map(_.text)
+      .via(countTopItemsZIO(_))
+      .foreach{ a => ZStream.fromIterable(a).foreach{ case (elem: String, count: Int) => putStrLn((count, elem).toString) } }
+      .as(ExitCode.success)
+      .orDie
+  }
+
+  /**
+    * Can be any of your own XML byte stream
+    */  
+  private val wikipediaXmlBytes: ZStream[Blocking, IOException, Byte] = {
+    val is: ZIO[Blocking, IOException, InputStream] = blocking.effectBlockingIO {
```

Comment:
I'm wondering, is it because gzip is not available from ZIO, that you need to wrap it in a GZIPInputStream?
Is there a ZIO HTTP library you could use here? Might prove a bit cleaner and more natural to a ZIO user (I only use InputStream in my example as that was quite straightforward, but perhaps should have really went for the http4s client)

Reply:
Not yet https://github.com/zio/zio/issues/3759 I can update the example after it is complete, however it would still need to be manually applied vs autodetect like http4s provides